source/robot_model/src/Model.cpp

```diff
@@ -108,7 +100,11 @@ void Model::init_model() {
   }
   // remove universe and root_joint frame added by Pinocchio
   this->frames_ = std::vector<std::string>(frames.begin() + 2, frames.end());
-  this->init_qp_solver();
+
+  // define the QP solver
+  this->qp_solver_ = std::make_unique<QPSolver>(
```

Comment:
shouldn't this be `make_shared?`

Reply:
I know you had a discussion about it that I didn't follow but it should correspond to the type defined in the header (which is currently shared pointer)

Reply:
As I discussed with @eeberhard, 

The robot model therefore should either:
* Explicitly dereference and copy the solver in a copy constructor,
* Hold the solver as an instance and not shared pointer,
* Forbid copy operations

I have implemented the first option, with that, I added a non-default copy constructor for robot model and within that copy constructor, I have explicitly dereferenced and copy the solver.